jdi-light/src/main/java/com/epam/jdi/light/elements/base/JDIBase.java

```diff
@@ -192,98 +226,31 @@ public JDIBase setName(String name) {
+    @JDebug
     public WebElement getWebElement() {
-        return get(new Object[]{});
+        return ELEMENT.getElementWithArgs.execute(this, new Object[]{});
     }
+
+    // Get Web Element using Search Rules and wait
+    @JDebug
     public WebElement get() {
-        WebElement element = get(new Object[]{});
-        VALIDATE_FOUND_ELEMENT.execute(this, element);
-        return element;
-    }
-    public WebElement get(Object... args) {
-        return ELEMENT.getElementWithArgs.execute(this, args);
-    }
-    public static JAction2<JDIBase, WebElement> VALIDATE_FOUND_ELEMENT = JDIBase::validateFoundElement;
-    private static void validateFoundElement(JDIBase base, WebElement element) {
-        for (JFunc1<WebElement, Boolean> rule : base.searchRules().values())
-            if (!rule.execute(element))
-                throw exception("Search rules failed for element. Please check base().searchRules() for element or in global settings(JDISettings.ELEMENT.searchRule)");
+        return get(new Object[]{});
     }
-    protected JFunc<WebElement> getElementFunc = null;
     public JDIBase setGetFunc(JFunc<WebElement> func) {
         getElementFunc = func;
         return this;
     }
-    private WebElement purify(WebElement element) {
-        return isInterface(element.getClass(), IBaseElement.class)
-            ? ((IBaseElement)element).base().get()
-            : element;
-    }
-    private UIElement getUIElementByLocator(WebElement element) {
-        return new UIElement(element).find(getLocator()).setup(b -> {
-            b.searchRules = searchRules;
-            b.timeout = timeout;
-        });
-    }
-    UIElement getWebListFromArgs(Object... args) {
-        if (locator.argsCount() == 0 && args.length == 1) {
-            if (args[0].getClass() == String.class)
-                return new WebList(this).get(args[0].toString());
-            if (isClass(args[0].getClass(), Enum.class))
-                return new WebList(this).get(getEnumValue((Enum<?>)args[0]));
-        }
-        throw exception("Can't get element with template locator '%s'. Expected %s arguments but found %s", getLocator(), locator.argsCount(), args.length);
-    }
-    WebElement getCachedElement() {
-        WebElement element = purify(webElement.get());
-        try {
-            element.getTagName();
-            if (locator.isNull())
-                return element;
-            return purify(getUIElementByLocator(element));
-        } catch (Exception ignore) {
-            if (getElementFunc == null) {
-                webElement.clear();
-                return null;
-            } else {
-                return webElement.set(purify(getElementFunc.execute()));
-            }
-        }
+    @JDebug
+    public WebElement get(Object... args) {
+        return ELEMENT.getElementAndValidate.execute(this, args);
     }
-    public Boolean strictSearch;
+
+    protected Boolean strictSearch;
```

Comment:
Codacy found an issue: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/gh/jdi-testing/jdi-light/pullRequest?prid=6397461)